datasets/dataset_imdb.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+"""
+IMDB dataset.
+inspired from:
+https://keras.io/examples/nlp/text_classification_from_scratch/
+https://builtin.com/data-science/how-build-neural-network-keras
+"""
+
+import tensorflow as tf
+import numpy as np
+from tensorflow import keras
+from keras.utils import to_categorical
+from keras import models
+from keras import layers
+from keras.datasets import imdb
+
+nb_words = 3000
+(x_train, y_train), (x_test, y_test) = imdb.load_data(num_words=nb_words)
+# Argument num_words => only the num_words first more frequent words are used
+# The others are replaced by 2
+# A Review start by the number 1
+# The longest review is less than 3000 words long
+
+def preprocess_dataset_inputs(data):
+    data_list = []
+    for sample in data:
+        sample += [2] * (nb_words - len(sample)) # 2 is the over_of_vocabulary number
```

Comment:
The words used in the reviews are replaced by integer from 3 to ... , the words 3 ('a') is the most frequent word in the whole dataset, the following is 4 ('of').
1 is the first integer of every review, used in the pre-standardization of the raw text in tf.keras.
2 is used to replaced a word that we don't want to take into account, and I used it to obtain same length vectors.
I don't know if it biasing the model created, because we also have to pre-process the sentence that the model will analyze.
The `load_data()` function can be used to replaced the most and the less frequent words by the `out_of_vocabulary` number. Maybe I should use it?
It seems very complicated: how should I know which word (based on its frequency) is relevant?

Reply:
ah ok. Sorry I had not read all [the documentation of the `load_data` function](https://keras.io/api/datasets/imdb/#loaddata-function) and, the beginning saying "3" codes for the third most present word, it miss-led me: I thought "1" was coding for the most used word "2" was for the second most used and so on. You can forget my comment. 

Though, yes, it seems interesting to filter the most common words as they do not contains much info. You can use the `get_word_index` function to identify which index is associated to which word (see [the documentation](https://keras.io/api/datasets/imdb/#getwordindex-function)) .

Reply:
I wrote a mistake: [1,2,3,4] code for [the,and,of,a]
The best things to do seems to use 0, as the out_of_vocab, and replaced the most and the less frequent  by 0.
Is it a mistake to use 0? As 0 isn't in the dictionary.

Here's the 100 most common words: `['the', 'and', 'a', 'of', 'to', 'is', 'br', 'in', 'it', 'i', 'this', 'that', 'was', 'as', 'for', 'with', 'movie', 'but', 'film', 'on', 'not', 'you', 'are', 'his', 'have', 'he', 'be', 'one', 'all', 'at', 'by', 'an', 'they', 'who', 'so', 'from', 'like', 'her', 'or', 'just', 'about', "it's", 'out', 'has', 'if', 'some', 'there', 'what', 'good', 'more', 'when', 'very', 'up', 'no', 'time', 'she', 'even', 'my', 'would', 'which', 'only', 'story', 'really', 'see', 'their', 'had', 'can', 'were', 'me', 'well', 'than', 'we', 'much', 'been', 'bad', 'get', 'will', 'do', 'also', 'into', 'people', 'other', 'first', 'great', 'because', 'how', 'him', 'most', "don't", 'made', 'its', 'then', 'way', 'make', 'them', 'too', 'could', 'any', 'movies']`
The best is to stop right before the 36th word: 'like'

Reply:
> This branch could be merge with #176 (and maybe #171 also)

@GauthierGar :

- Usually we don't consider merging two open PRs. It makes things simpler to consider PRs independently
- #171 is an issue, that is a given topic that we want to work on. A PR addresses an issue (or several), fixes it. But merging a PR with an issue wouldn't mean much :)

Reply:
This depends upon the arguments you pass to `load_data()`, like in `imdb.load_data(num_words=10000, skip_top=35, start_char=1, oov_char=2, index_from=3)`.

Reply:
I have completely change this function
The new pre-processing of inputs is quite arbitrary but give good results.
We only keep the 10 000 more frequent words (the others are replaced by the out_of_vocabulary integer = 2), and the 200 first words of the reviews.
If the `len(review) < 200` we add 0 at the start to obtain a review of 200 words.